vendor-extra/LiberoPageBundle/src/EventListener/BuildView/ItemListListener.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Libero\LiberoPageBundle\EventListener\BuildView;
+
+use ArrayAccess;
+use DOMNodeList;
+use FluentDOM\DOM\Element;
+use Libero\ViewsBundle\Views\ContextAwareTranslation;
+use Libero\ViewsBundle\Views\LazyView;
+use Libero\ViewsBundle\Views\SimplifiedViewConverterListener;
+use Libero\ViewsBundle\Views\TemplateView;
+use Libero\ViewsBundle\Views\View;
+use Libero\ViewsBundle\Views\ViewConverter;
+use Symfony\Contracts\Translation\TranslatorInterface;
+use function array_filter;
+use function array_map;
+use function count;
+use function Libero\ViewsBundle\array_has_key;
+
+final class ItemListListener
+{
+    use ContextAwareTranslation;
+    use SimplifiedViewConverterListener;
+
+    private $converter;
+
+    public function __construct(ViewConverter $converter, TranslatorInterface $translator)
+    {
+        $this->converter = $converter;
+        $this->translator = $translator;
+    }
+
+    protected function handle(Element $object, TemplateView $view) : View
+    {
+        /** @var DOMNodeList<Element> $itemRefs */
+        $itemRefs = $object('libero:item-ref');
+
+        $level = $view->getContext()['level'] ?? 1;
+        if ($view->hasContext('list_title')) {
+            $level++;
+            $view = $view->withArgument(
+                'title',
+                [
+                    'level' => $view->getContext()['level'] ?? 1,
+                    'text' => $this->translate($view->getContext('list_title'), $view->getContext()),
+                ]
+            );
+        }
+
+        if (0 === count($itemRefs)) {
+            if (!$view->hasContext('list_empty')) {
+                return $view->withArgument('list', []);
+            }
+
+            return $view->withArgument(
+                'list',
+                ['empty' => $this->translate($view->getContext('list_empty'), $view->getContext())]
+            );
+        }
+
+        $items = [];
+        foreach ($itemRefs as $itemRef) {
+            $items[] = $this->converter->convert(
+                $itemRef,
+                '@LiberoPatterns/teaser.html.twig',
+                ['level' => $level] + $view->getContext()
+            );
+        }
+
+        return new LazyView(
+            function () use ($view, $items) {
+                return $view->withArgument(
+                    'list',
+                    [
+                        'items' => array_filter(
+                            array_map(
+                                function (View $view) {
+                                    if (!$view instanceof ArrayAccess) {
+                                        return [];
```

Comment:
what does this case cover? It looks like we are skipping an item, but don't know why we would have an empty element here

Reply:
Can move the check higher up. `ViewConverter::convert` could return a `StringView`. Might want to throw an exception though, or at least log it.